### PR TITLE
Switch to using ember-cp-validations

### PR DESCRIPTION
**Pull request by spruce**
## Why are we switching?

As discussed in #139 the currently used solution is no longer supported by their maintainers, that's why I think it is time to finally switch to a supported solution. (Nice Side-effect: we finally can support server-side errors)

## Which version will the change be happening?

In Version 1.0 (hopefully released before the end of this year) we will still be supporting ember-validations and ember-cp-validations (via a mixin)

In Version 2.0 we are gonna drop support for ember-validations. This version is gonna be released by june 2017 (hopefully) 

## What still needs to be done:
- [x] make sure tests work correctly with ECV
- [x] make sure ECV works together with computed props on model
- [ ] update version number
- [x] make it work without ember-data

## Further questions

If anything is unclear or you have a general question feel free to ask.

### CODE REVIEW

File: tests/dummy/app/mixins/ember-cp-validations-helper.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+//import Ember from 'ember';
```
Comment (GCorbel) [inline tests/dummy/app/mixins/ember-cp-validations-helper.js]: I Think we can remove this comment

File: tests/dummy/app/models/person-with-cp-validations.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    validator('presence', true),
+    validator('length', {
+      min: 6
+    })
+  ],
+  comment: validator('presence', true),
+  gender: validator('presence', true),
+});
+
+
```
Comment (GCorbel) [inline tests/dummy/app/models/person-with-cp-validations.js]: To many spaces

File: tests/unit/mixins/ember-cp-validations-helper-test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import Ember from 'ember';
+import EmberCpValidationsHelperMixin from 'ember-rapid-forms/mixins/ember-cp-validations-helper';
+import { moduleFor, test } from 'ember-qunit';
+
+moduleFor('mixin:ember-cp-validations-helper', 'Unit | Mixin | ember cp validations helper');
+
+// Replace this with your real tests.
```
Comment (GCorbel) [inline tests/unit/mixins/ember-cp-validations-helper-test.js]: Remove this line

Comment (GCorbel) [inline tests/unit/mixins/ember-cp-validations-helper-test.js]: Maybe more tests are needed

File: addon/mixins/ember-cp-validations-helper.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    const isEmpty = modelErrors.get('isEmpty');
+
+    if (wasEmpty && !isEmpty) {
+      modelErrors.trigger('becameInvalid');
+    } else if (!wasEmpty && isEmpty) {
+      modelErrors.trigger('becameValid');
+    }
+  }
+});
```
Comment (GCorbel) [inline addon/mixins/ember-cp-validations-helper.js]: A refactoring or more comments can be useful to make it clearer

File: tests/unit/mixins/ember-cp-validations-helper-test.js (hunk 3 above)
Comment (spruce) [inline tests/unit/mixins/ember-cp-validations-helper-test.js]: Yes definitely more tests are needed. But I've got no good idea what they should test. Maybe an object with some fake validations on it an whether they will correctly render. But this should be integration tests, no?

**Comment by GCorbel** [comment]
About tests, I think you can just check if things are correctly binded. Something like `model.validate` fills `model.errors`. What you think ?

**Comment by spruce** [comment]
Ok, I will look into adding some tests like that.

**Comment by spruce** [comment]
How do you feel about the just added test?

**Comment by GCorbel** [comment]
A did little commit. It looks good, we can merge it.

**Comment by GCorbel** [comment]
Some page are failing now. We can't start the application because some models use 'ember-validations'. I will correct it.

**Pull request merged**